packages/perseus/src/__tests__/server-item-renderer.test.tsx

```diff
@@ -288,16 +288,22 @@ describe("server item renderer", () => {
         // render tree created.
         // Finally we re-render and poke the asset status to loaded. At that
         // everything is loaded.
+        //
+        // TODO: is it possible to use an existing widget to test the same logic?
```

Comment:
**thought**: I think this mocked asset loading widget might be one we want to keep.  The `image` widget would be a widget that hooks into the asset loading context, but it may be tricky to assert that the ServerItemRenderer does the right thing using the implementation of a production widget, and not risk regression. 

The implementation of that widget is extremely small. Also, I wonder if we could eliminate all the `@ts-expect-error` suppressions by simply adding the mock widget to the list of widgets, but mark it as `hidden: true`. Maybe that's a net negative though. 

What do you think?

Reply:
I came to the same conclusion as you: keep the widget. I chose to mark it as deprecated though so that people would at least be hesitant to use it (I personally think we should bias towards using real widgets when possible); that meant that I had to suppress "no deprecated" errors when using it.

I'd be fine removing the TODOs and deprecated stuff if you think that's best.